### DISCUSSION TITLE
Welcome to galapagos Discussions!

### DISCUSSION THREAD

**Discussion by albrechtflo-hg** [HermesGermany/galapagos#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by theesen** [comment]
Might be a stupid question, but I couldn't understand from the README or find another source of information. What exactly is Galapagos doing? We are operating Kafka Clusters as well and always look for ways to improve our setup and automation for it. Which is why I am looking how other companies approach this topic.

**Comment by albrechtflo-hg** [comment]
Hi @theesen , yes, we definitely can improve on demonstrating what Galapagos is (and what it is not).

Main features / use cases of Galapagos:

* Enable the teams in your company to create "public" topics for their applications
* Enable other teams to search for "public" topics they are interested in, and acquire required access rights for their applications
* Enforce some guidelines around topics and their contents, to enforce an "Event Driven Architecture" using Kafka (publish events, not data objects)
* Enforce consumer-compatibility of JSON Schema changes (for the contents of "public" topics)
* Cleanly stage topics, schemas, subscriptions to later stages
* Deprecation of topics

In short, it helps you keeping an overview of "who talks with whom" and letting teams self-provision what they need, without running into "Kafka chaos".

Galapagos does **not** help very much with automation. Although Galapagos is developed "API-first" an can completely be used via REST API, it does usually not make much sense to do so, as you usually create your topic **once** on the DEV cluster, subscribe **once** to a required public topic etc.

Even when publishing new JSON schema versions, it is totally fine to **first** start producing the contents adhering to the new schema and **then** update the schema, as the consumer compatibility guarantees that "old" consumers must be able to handle the new schema as well (e.g. by ignoring added fields).

Also, Galapagos assumes few central Kafka clusters (one central cluster per stage) and is **not** intended e.g. for setups where every team operates their own cluster(s). This would be contrary to our approach to an enterprise-wide Event Driven Architecture.

I hope that this helps in getting a first overview - feel free to add more questions. There really is no such thing as a "stupid question" here :-)

Regards,
Florian

**Reply by theesen** [reply]
Thank you for clarifying. I'll try it out in the next days.
